skyportal/handlers/api/photometry.py

```diff
@@ -50,6 +50,9 @@ def save_data_using_copy(rows, table, columns):
     # Prepare data
     output = StringIO()
     df = pd.DataFrame.from_records(rows)
+    # Coerce missing non-numbers and numbers, respectively, for SQLAlchemy
+    df.replace("NaN", "null", inplace=True)
+    df.replace(np.nan, "NaN", inplace=True)
```

Comment:
I'm not sure the order of inputs is correct. 
Unless I'm confused about this, it would turn "NaN"->"null" and then np.nan to "NaN".

Reply:
The order matter. NaN (numbers) are expressed as np.nan up until this point but need to be "NaN" for SQLA. NaN (strings) are expressed as NaN but need to be "null" for SQLA. It took me awhile to get here...

Reply:
OK I didn't realize there's a distinction between `np.nan` and `"NaN"` in the dataframes we are ingesting. 
This is really confusing. 
What is the difference between "null" and "NaN" in the photometry in the end?

Reply:
There's no conceptual difference between null, None, and NaN from a photometry/flux perspective, but `pandas`, starting in 1.3.*, started treating these differently internally such that the old code broke (in dozens of tests) when we went from 1.2.5 to 1.3.0.  This PR fixes this.

Reply:
The only thing I am not clear on is why you want to end up in SQLA with both "NaN" and "null"?

Reply:
that happens elsewhere in the code, using pandas to read in (user) uploaded CSV files.

Reply:
So why not convert all the different user input and store only "NaN" in the database?